### PR TITLE
feat(list): add filterProps property to specify which properties to filter against

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9619

## Summary

- add filterProps property to list
- changes `matchFields` to `filterProps` on the filter component.
- add tests

### CODE REVIEW

File: packages/calcite-components/src/components/list/list.e2e.ts [hunk 1]
```diff
@@ -373,6 +377,51 @@ describe("calcite-list", () => {
...
+          value="value-4"
+        ></calcite-list-item>
+      </calcite-list>
+    `);
+
+    await page.waitForChanges();
+    const list = await page.find("calcite-list");
+    list.setProperty("matchFields", ["label", "description"]);
+    await page.waitForChanges();
+    await page.waitForTimeout(listDebounceTimeout);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.e2e.ts:415]: Is there an event we can listen for? This is fine for now, but we should start moving away from timeouts as [`puppeteer@22.2.0` removed `page.waitForTimeout`](https://github.com/puppeteer/puppeteer/blob/7966dd726288bda48650a3604f8e12c6e4be41db/packages/puppeteer-core/CHANGELOG.md#2200-2024-02-05).

File: packages/calcite-components/src/components/list/list.e2e.ts [hunk 2]
```diff
@@ -373,6 +377,51 @@ describe("calcite-list", () => {
     expect(visibleItems.map((item) => item.id)).toEqual(["label-match", "description-match", "value-match"]);
   });
 
+  it("filters initially with matchFields", async () => {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.e2e.ts:380]: Do we need a test for user-interaction filtering? Under the hood, both work with `filterText`, so probably not?

File: packages/calcite-components/src/components/list/list.tsx [hunk 3]
```diff
@@ -160,6 +160,16 @@ export class List
    */
   @Prop({ reflect: true }) loading = false;
 
+  /**
+   * Specifies the fields to match against when filtering. If not set, all fields will be matched.
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.tsx:164]: Can we add a note about the default fields being used for filtering (`label`, `description`, `metadata`, `value`)? cc @geospatialem

**Review by jcfranco — APPROVED** [review]
📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝
📝🏑📝📝📝🏑📝📝🏑🏑📝📝🏑🏑🏑📝📝🏑🏑🏑📝🏑🏑🏑🏑📝🏑📝
📝🏑🏑📝📝🏑📝🏑📝📝🏑📝📝🏑📝📝🏑📝📝📝📝🏑📝📝📝📝🏑📝
📝🏑📝🏑📝🏑📝🏑📝📝🏑📝📝🏑📝📝🏑📝📝📝📝🏑🏑🏑📝📝🏑📝
📝🏑📝📝🏑🏑📝🏑📝📝🏑📝📝🏑📝📝🏑📝📝📝📝🏑📝📝📝📝📝📝
📝🏑📝📝📝🏑📝📝🏑🏑📝📝🏑🏑🏑📝📝🏑🏑🏑📝🏑🏑🏑🏑📝🏑📝
📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝📝

File: packages/calcite-components/src/components/list/list.e2e.ts (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/components/list/list.e2e.ts:415]: Not currently, we would need some kind of event for when filteredItems is updated. Can you create an issue to add this so we can clean up these tests?

File: packages/calcite-components/src/components/list/list.e2e.ts (hunk 2 above)
Comment (driskull) [reply packages/calcite-components/src/components/list/list.e2e.ts:380]: Probably not necessary.

File: packages/calcite-components/src/components/list/list.tsx (hunk 3 above)
Comment (driskull) [reply packages/calcite-components/src/components/list/list.tsx:164]: Yep. will do

File: packages/calcite-components/src/components/list/list.e2e.ts (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/list/list.e2e.ts:415]: 🔥https://github.com/Esri/calcite-design-system/issues/9661🔥

File: packages/calcite-components/src/components/list/list.tsx [hunk 4]
```diff
@@ -160,6 +160,16 @@ export class List
    */
   @Prop({ reflect: true }) loading = false;
 
+  /**
+   * Specifies the fields to match against when filtering. If not set, all fields will be matched (label, description, metadata, value).
```
Comment (macandcheese) [inline packages/calcite-components/src/components/list/list.tsx:164]: Sorry, late to the party here - is "fields" an obvious term here for this? "If not set, all fields will be matched (label, description, metadata, value)"

Really it's matching supplied property values? Feel free to disregard but "fields" doesn't immediately convey what will be matched to me.

Comment (driskull) [reply packages/calcite-components/src/components/list/list.tsx:164]: We already called the property `matchFields` on `filter.tsx` but we can change before we release if we have a beter name. `matchPropertyValues` or `matchKeyValues`? I think its really a key. Could also be `matchKeys`?

@jcfranco

Comment (driskull) [reply packages/calcite-components/src/components/list/list.tsx:164]: I kinda like `filterProps`. // the properties to use for filtering? @macandcheese? @jcfranco

Comment (driskull) [reply packages/calcite-components/src/components/list/list.tsx]: We will likely need to update the changelog at release to modify matchFields to filterProps on the filter component.

**Pull request merged**